CalculateHash

SupinePandora43 opened the pull request:
`CalculateHash` function example

SupinePandora43 committed:
initial

SupinePandora43 committed:
NativeMath actions

SupinePandora43 committed:
net5

SupinePandora43 committed:
set NativeMath to nil

SupinePandora43 committed:
free NativeMath using GCHandle

SupinePandora43 committed:
lua tests

SupinePandora43 committed:
math.Approach

SupinePandora43 committed:
donothingwithit

SupinePandora43 committed:
handles list

SupinePandora43 committed:
current

SupinePandora43 committed:
thousand

SupinePandora43 committed:
current

SupinePandora43 committed:
Distance

SupinePandora43 committed:
disable jit

SupinePandora43 committed:
enable JIT
use static function pointer
allow unsafe blocks

SupinePandora43 committed:
HashCalc

SupinePandora43 committed:
ToString("X2")

SupinePandora43 committed:
print

SupinePandora43 committed:
builder

SupinePandora43 committed:
return 1

SupinePandora43 committed:
fix casing

SupinePandora43 committed:
sha256 unicode test

SupinePandora43 committed:
CalculateHash

SupinePandora43 committed:
md5 test

SupinePandora43 committed:
SHA512

GlebChili commented on .github/workflows/test.lua:
```diff
@@ -0,0 +1,41 @@
+	local sha256 = CalculateHash("SHA256","test")
+	assert(sha256==string.upper("9f86d081884c7d659a2feaa0c55ad015a3bf4f1b2b0b822cd15d6c15b0f00a08"))
+
+	local sha256_unicode = CalculateHash("SHA256","☭")
```
We should avoid using political or ideological symbols in code.

GlebChili commented on .github/workflows/test.lua:
```diff
@@ -0,0 +1,41 @@
+	local md5 = CalculateHash("MD5","test")
+	assert(md5==string.upper("098f6bcd4621d373cade4e832627b4f6"))
+
+	local md5_unicode = CalculateHash("MD5","☭")
```
We should avoid using political or ideological symbols in code.

GlebChili commented on .github/workflows/test.lua:
```diff
@@ -0,0 +1,41 @@
+	local sha512 = CalculateHash("SHA512","test")
+	assert(sha512==string.upper("ee26b0dd4af7e749aa1a8ee3c10ae9923f618980772e473f8819a5d4940e0db27ac185f8a0e1d5f84f88bc887fd67b143732c304cc5fa9ad8e6f57f50028a8ff"))
+
+	local sha512_unicode = CalculateHash("SHA512","☭")
```
We should avoid using political or ideological symbols in code.

GlebChili commented on HashCalc/HashCalc.cs:
```diff
@@ -0,0 +1,60 @@
+			handles = new();
+
+			lua.PushSpecial(SPECIAL_TABLES.SPECIAL_GLOB);
+			unsafe
+			{
+				lua.PushCFunction(&CalculateHash);
+				lua.SetField(-2, "CalculateHash");
+			}
```
I think it is a bad idea to use PushCFunction for the first example. It is really unsafe and must be used with caution. This use case should be a subject of special tutorial later. Use PushManagedFunction instead.

GlebChili commented on .editorconfig:
```diff
@@ -0,0 +1,9 @@
+[*]
+end_of_line              = crlf
+charset                  = utf-8
+indent_style             = tab
+indent_size              = tab
+trim_trailing_whitespace = true
+[*.yml]
+indent_style = space
+indent_size  = 2
```
Do we really need an .editorconfig in the Example? I think imposing code style on end users is a bad idea.

SupinePandora43 committed:
Managed function

SupinePandora43 committed:
remove unused includes
delete .editorconfig
replace ~~ussr~~ hammer and sickle unicode with heart unicode

GlebChili commented on HashCalc/HashCalc.cs:
```diff
@@ -0,0 +1,53 @@
+
+		private static int CalculateHash(ILua lua)
+		{
+			string hashName = lua.GetString(1);
+			string data = lua.GetString(2);
```
I think we have to check that function parameters are valid i.e. we have at lease two params passed, both parameters are strings (or can be automatically converted to strings), the hash algorithm name is valid, etc.

SupinePandora43 committed:
add hashName check

SupinePandora43 committed:
add timeout minutes
